### DISCUSSION TITLE
Calendar Versioning (CalVer) Support

### DISCUSSION THREAD

**Discussion by x80486** [goreleaser/goreleaser#4980] [Ideas, Issue Triage, and General discussions]
I would like to explore your opinion to consider adding support for an alternative versioning scheme: calendar versioning. Semantic versioning is well-suited for certain development workflows, but calendar versioning offers significant advantages for other scenarios, especially for projects where release dates and schedules are critical.

I use [this](https://github.com/k1LoW/calver) library. It's uncertain to me if this could be used or not, but it works like a charm, so I was wondering if that's something you would consider as a built-in feature for `GoReleaser`.

Alternatively, is there a way to integrate/use this somehow with `GoReleaser`?

**Comment by caarlos0** [comment]
it should work as is, as the calver is `number.number.number`, just like semver.

have you find any problems?

**Reply by x80486** [reply]
No it works correctly with that scheme as well :ok_hand:

I've got _distracted_ with some other configuration settings while trying to set up `GoReleaser`, so I thought the tool didn't support that.

**Comment by axgkl** [comment]
works when  it is 'v2025.2.15', fails on  'v2025.02.15'. No big deal but one has to take care of that when gen-ing the release tag.

**Reply by caarlos0** [reply]
ahh, right, semver spec doesn't allow preceding zeroes.

I guess we could specifically support calver maybe something like this https://github.com/goreleaser/goreleaser/pull/5568 ?

**Reply by axgkl** [reply]
lgtm

**Reply by h3adache** [reply]
Is this something that recently? I was publishing packages with leading zeros up until Sept of last year with running the latest.
Examples, these all published fine.
```
v2024.07.03
v2024.09.26
```

but this failed today.
```
v2025.04.21
```

**Reply by caarlos0** [reply]
It might be that the semver lib started properly handling it in this timeframe, but I'm not sure and afk now
